room_assistance/indico_room_assistance/migrations/20190220_1138_b160812c186b_add_startup_assistance_requests_table.py

```diff
@@ -0,0 +1,100 @@
+"""Add room assistance requests table
+
+Revision ID: b160812c186b
+Revises:
+Create Date: 2019-02-20 11:38:08.691528
+"""
+
+from __future__ import print_function, unicode_literals
+
+from alembic import context, op
+from terminaltables import AsciiTable
+
+
+# revision identifiers, used by Alembic.
+revision = 'b160812c186b'
+down_revision = None
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    if context.is_offline_mode():
+        raise Exception('This upgrade is only possible in online mode')
+
+    conn = op.get_bind()
+    conn.execute('''
+        INSERT INTO events.requests (event_id, type, state, data, created_by_id, created_dt)
+        SELECT
+            rvl.linked_event_id,
+            'room-assistance',
+            1,
+            json_build_object('start_dt', rv.start_dt, 'reason', ''),
+            (SELECT id FROM users.users WHERE is_system),
+            rv.start_dt
+        FROM roombooking.reservations rv
+        JOIN roombooking.reservation_links rvl ON rv.link_id = rvl.id AND rvl.link_type = 2
+        WHERE rv.needs_assistance
+    ''')
+
+    invalid_reservations = conn.execute('''
+        SELECT
+            rv.id,
+            rv.room_id,
+            format('%%s/%%s-%%s', r.building, r.floor, r.number),
```

Comment:
Why double `%`?

Reply:
To escape `%` so that it is not interpreted as a part of a formatted string.

Reply:
In Python? Unless you use `%` after the string, you do not need to escape it..

Reply:
Since I first thought "why, there's no `%` formatting applied on the string": `%s` is also used by the Postgres DBAPI for params.

Reply:
It's strange because I don't need to escape it in my script :thinking:

Reply:
I guess you aren't passing it to the DBAPI since you are using the models?